src/test/kotlin/ua/marchenko/artauction/config/annotation/customScheduled/CustomScheduledAnnotationTest.kt

```diff
@@ -0,0 +1,117 @@
+import config.annotation.customScheduled.CustomScheduledTestServiceWithoutAnnotationImpl
+import getRandomString
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.slot
+import io.mockk.verify
+import java.time.Duration
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.config.BeanDefinition
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory
+import org.springframework.context.ApplicationContext
+import org.springframework.context.event.ContextRefreshedEvent
+import ua.marchenko.artauction.common.annotation.scheduled.contextListener.CustomScheduledInvokerContextListener
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.isAccessible
+import kotlin.reflect.jvm.javaField
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.slf4j.LoggerFactory
+
+class CustomScheduledAnnotationTest {
+
+    @MockK
+    private lateinit var beanFactory: ConfigurableListableBeanFactory
+
+    @MockK
+    private lateinit var context: ApplicationContext
+
+    @MockK
+    private lateinit var scheduledExecutorService: ScheduledExecutorService
+
+    @MockK
+    private lateinit var beanDefinition: BeanDefinition
+
+    @InjectMockKs
+    private lateinit var customScheduledInvokerContextListener: CustomScheduledInvokerContextListener
+
+    private lateinit var testLogAppender: ListAppender<ILoggingEvent>
+
+    @BeforeEach
+    fun setUp() {
+        val field = customScheduledInvokerContextListener::class.declaredMemberProperties
+            .first { it.name == SCHEDULED_EXECUTOR_SERVICE_NAME }
+        field.isAccessible = true
+        field.javaField?.set(customScheduledInvokerContextListener, scheduledExecutorService)
+
+        val logger = LoggerFactory.getLogger(CustomScheduledInvokerContextListener::class.java) as Logger
+        testLogAppender = ListAppender()
+        testLogAppender.start()
+        logger.addAppender(testLogAppender)
+    }
+
+    @Test
+    fun `should schedule method when CustomScheduled annotation is present`() {
+        // GIVEN
+        val beanName = getRandomString()
+        val beanNames = arrayOf(beanName)
+        val testBean = CustomScheduledTestServiceWithAnnotationImpl()
+        val delaySlot = slot<Long>()
+
+        every { context.beanDefinitionNames } returns beanNames
+        every { beanFactory.getBeanDefinition(beanName) } returns beanDefinition
+        every { beanDefinition.beanClassName } returns CustomScheduledTestServiceWithAnnotationImpl::class.java.name
+        every { context.getBean(beanName) } returns testBean
+        every { scheduledExecutorService.scheduleWithFixedDelay(any(), any(), any(), TimeUnit.SECONDS) } returns null
```

Comment:
Seems like integration test, honestly
Mocking Spring Boot parts is not great idea